PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/component/ApplicationInsightsPanel.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright (c) Microsoft Corporation
+ *
+ * All rights reserved.
+ *
+ * MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+ * documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+ * to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of
+ * the Software.
+ *
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.microsoft.intellij.runner.functions.component;
+
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.ui.PopupMenuListenerAdapter;
+import com.intellij.ui.SimpleListCellRenderer;
+import com.microsoft.azure.management.applicationinsights.v2015_05_01.ApplicationInsightsComponent;
+import com.microsoft.intellij.common.CommonConst;
+import com.microsoft.tooling.msservices.helpers.azure.sdk.AzureSDKManager;
+import io.reactivex.rxjava3.core.Observable;
+import io.reactivex.rxjava3.disposables.Disposable;
+import io.reactivex.rxjava3.schedulers.Schedulers;
+import org.jetbrains.annotations.NotNull;
+
+import javax.swing.*;
+import javax.swing.event.PopupMenuEvent;
+import java.io.InterruptedIOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class ApplicationInsightsPanel extends JPanel {
+    private static final String CREATE_NEW_APPLICATION_INSIGHTS = "Create New Application Insights...";
+
+    private JComboBox cbInsights;
+    private JPanel pnlRoot;
+    private String subscriptionId;
+
+    private Disposable rxDisposable;
+    private ApplicationInsightsWrapper selectWrapper;
+    private ApplicationInsightsWrapper newInsightsWrapper;
+
+    public ApplicationInsightsPanel() {
+        cbInsights.setRenderer(new SimpleListCellRenderer() {
+            @Override
+            public void customize(@NotNull final JList list,
+                                  final Object o,
+                                  final int i,
+                                  final boolean b,
+                                  final boolean b1) {
+                if (o != null) {
```

Comment:
Are there any chance of null o? If there is any possibility, when o is null, shall we setText("")?

Reply:
Usually not, i'll add the empty text for null in case some corner case